contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.h

```diff
@@ -57,6 +57,26 @@ class CryptoMbContext {
   Event::SchedulableCallbackPtr schedulable_{};
 };
 
+// CryptoMbEcdsaContext is a CryptoMbContext which holds the extra ECDSA parameters and has
+// custom initialization function.
+class CryptoMbEcdsaContext : public CryptoMbContext {
+public:
+  CryptoMbEcdsaContext(bssl::UniquePtr<EC_KEY> ec_key, Event::Dispatcher& dispatcher,
+                       Ssl::PrivateKeyConnectionCallbacks& cb)
+      : CryptoMbContext(dispatcher, cb), ec_key_(std::move(ec_key)) {}
+  bool ecdsaInit(const uint8_t* in, size_t in_len);
+
+  // ECDSA key.
+  bssl::UniquePtr<EC_KEY> ec_key_{};
+  // ECDSA context to create the ephemeral key k_.
+  bssl::UniquePtr<BN_CTX> ctx_{};
+  BIGNUM* k_{};
+  // ECDSA parameters, which will contain values whose memory is managed within
+  // BoringSSL ECDSA key structure, so not wrapped in smart pointers.
+  const BIGNUM* priv_key_{};
+  size_t sig_len_{};
```

Comment:
There is a `out_len_` in the `CryptoMbContext`, and it seems same purpose with `sig_len_`, is it possible using the `out_len_` here?

Reply:
No. Through the `sig_len_` is a fixed length of signature, the marshaled signature size `out_len_` may be different.

https://github.com/envoyproxy/envoy/pull/31789/files#diff-38f294cd090527c4632a72082be002caf3fdbfca58ea37965190659058032d9fR581